src/riak_kv_vnode.erl

```diff
@@ -878,41 +839,104 @@ handle_command({get_index_entries, Opts},
+                                                                   mod=Mod,
+                                                                   modstate=ModState,
+                                                                   counter=CS,
+                                                                   vnodeid=VId}) ->
+    BackendStatus = {backend_status, Mod, Mod:status(ModState)},
+    #counter_state{cnt=Cnt, lease=Lease, lease_size=LeaseSize, leasing=Leasing} = CS,
+    CounterStatus = [{counter, Cnt}, {counter_lease, Lease},
+                     {counter_lease_size, LeaseSize}, {counter_leasing, Leasing}],
+    VNodeStatus = [BackendStatus, {vnodeid, VId} | CounterStatus],
+    {reply, {vnode_status, Index, VNodeStatus}, State};
+handle_request(kv_delete_request, Req, _Sender, State) ->
+    BKey = riak_kv_requests:get_bucket_key(Req),
+    do_delete(BKey, State);
+handle_request(kv_vclock_request, Req, _Sender, State) ->
+    BKeys = riak_kv_requests:get_bucket_keys(Req),
+    {reply, do_get_vclocks(BKeys, State), State}.
+
 
 %% @doc Handle a coverage request.
 %% More information about the specification for the ItemFilter
 %% parameter can be found in the documentation for the
 %% {@link riak_kv_coverage_filter} module.
-handle_coverage(?KV_LISTBUCKETS_REQ{item_filter=ItemFilter},
-                _FilterVNodes,
-                Sender,
-                State=#state{async_folding=AsyncFolding,
-                             bucket_buf_size=BufferSize,
-                             mod=Mod,
-                             modstate=ModState}) ->
+handle_coverage(Req, FilterVNodes, Sender, State) ->
+    handle_coverage_request(riak_kv_requests:request_type(Req),
+                            Req,
+                            FilterVNodes,
+                            Sender,
+                            State).
+
+handle_coverage_request(kv_listkeys_request, Req, FilterVNodes, Sender, State) ->
+    Bucket = riak_kv_requests:get_bucket(Req),
+    ItemFilter = riak_kv_requests:get_item_filter(Req),
+    ResultFun = case riak_kv_requests:get_ack_backpressure(Req) of
+                    true  -> result_fun_ack(Bucket, Sender);
+                    false -> result_fun(Bucket, Sender)
+                end,
+    Opts = [{bucket, Bucket}],
+    handle_coverage_keyfold(Bucket, ItemFilter, ResultFun,
+                            FilterVNodes, Sender, Opts, State);
+handle_coverage_request(kv_listbuckets_request,
+                        Req,
+                        _FilterVNodes,
+                        Sender,
+                        State=#state{async_folding=AsyncFolding,
+                                     bucket_buf_size=BufferSize,
+                                     mod=Mod,
+                                     modstate=ModState}) ->
+    ItemFilter = riak_kv_requests:get_item_filter(Req),
     %% Construct the filter function
     Filter = riak_kv_coverage_filter:build_filter(ItemFilter),
     BufferMod = riak_kv_fold_buffer,
```

Comment:
```
BufferMod = riak_kv_fold_buffer,
 
Buffer = BufferMod:new(BufferSize, result_fun(Sender)),
FoldFun = fold_fun(buckets, BufferMod, Filter, undefined),
FinishFun = finish_fun(BufferMod, Sender),
```
while the use in the FoldFun and FinishFun is fine., it is problematic in the BufferMod:new/3 since it stops xref from spotting the call. 
Please replace that one with the actual atom.